shlr/tcc/tccgen.c

```diff
@@ -1178,13 +1182,19 @@ static void struct_decl(CType *type, int u) {
 						{
 							int type_bt = type1.t & VT_BTYPE;
 							//eprintf("2: %s.%s = %s\n", ctype, name, varstr);
-							tcc_appendf ("[+]%s.%s=%s\n",
-								ctype, name, varstr);
-							tcc_appendf ("%s.%s.%s.meta=%d\n",
-								ctype, name, varstr, type_bt);
-							/* compact form */
-							tcc_appendf ("%s.%s.%s=%s,%d,%d\n",
-								ctype, name, varstr, b, offset, arraysize);
+							if (is_typedef && autonamed) {
+								tcc_typedef_appendf ("[+]typedef.%%1$s.fields=%s\n", varstr);
+								tcc_typedef_appendf ("typedef.%%1$s.%s.meta=%d\n", varstr, type_bt);
```

Comment:
why %%1$s ? wtf

Reply:
I'ts a bit wild, but it works. I put it this way because these fields need the typedef identifier later on.